File: Bisiesto.m
clc; clear all; close all;
pkg load database;
retry= true;

while retry
  try
    fprintf("\n Elija una opcion \n 1.Correr el programa \n 2.Mostrar Historial \n");
    m=(input('Elija:   '));
    if m>2||m==0|m<0
      fprintf("Elija una opcion correcta \n");
    endif
    if m==1
      a=input('Ingrese el año:  ');
      Es='';
      if (mod(a,400)==0)||(mod(a,4)==0)||(mod(a,100)==0)
         Es='Bisiesto';
         fprintf("El año es Bisiesto \n");
      elseif
         Es='No es bisiesto'; 
         fprintf("El año no es bisiesto \n");
      endif
      params= cell(1,2);
      params{1,1}=a;
      params{1,2}=Es;
      conn = pq_connect(setdbopts('dbname','HWP','host','localhost',
      'port','5432','user','postgres','password','123fgthg'));
      N=pq_exec_params(conn, "insert into bisiesto(yer, bis) values($1,$2);",params); %insertar datos en la tabla
      retry=false;
    endif
    if m==2
      conn = pq_connect(setdbopts('dbname','HWP','host','localhost',
      'port','5432','user','postgres','password','123fgthg'));
      N=pq_exec_params(conn, "Select * from bisiesto;"); %insertar datos en la tabla
      disp(N)
      retry=false;
    endif
  catch
    fprintf("A ingresado un valor erroneo vuelva a intentarlo %d \n");
    msg = lasterror.message;
    fprintf(msg);
  end_try_catch
endwhile

File: bisiesto.py
import psycopg2
from tabulate import tabulate
def  post(x,y):
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "123fgthg",
            dbname = "HWP"
            )
    except:
        print("Sin Conexion Exitosa\n")
            
    cursor = conexion.cursor()
    cursor.execute("INSERT INTO bisiesto(yer, bis) VALUES (%s, %s);", (x, y))
    conexion.commit()
    cursor.close()
    conexion.close()

def ret():
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "123fgthg",
            dbname = "HWP"
            )
    except:
        print("Sin Conexion Exitosa\n")
    
    cursor = conexion.cursor()
    cursor.execute("SELECT * from bisiesto;")
    print(tabulate(cursor, headers=["ID", "AÑO", "BIS"], tablefmt="fancy_grid", numalign ="center"))

print("AÑOS BISIESTOS \n")
r=[]
while True:
    try:
        print("Elija una opcion:\n1.Correr el Programa\n2.Mostrar el Historial\n3.Salir")
        b=int(input("Elija:\n"))
        if b>3 or b<=0:
            print("Valor ingresado fuera del rango del menu.\n")
        elif b==1:
            o=int(input("Ingrese el año \n"))
            re =""
            if o % 4 == 0:
                if o % 100 == 0:
                    if o % 400 == 0:
                        print('El año es bisiesto')
                        re="Bisiesto"
                    else:
                        print('El año no es bisiesto')
                        re="No Bisiesto"
                else:
                    print('El año es bisiesto.')
                    re="Bisiesto"
            else:
                print('El año no es bisiesto.')
                re="No Bisiesto"
            post(o,re)
        elif b==2:
            ret()
        elif b==3:
            break

    except Exception as e:
        print("\n A ingresado un valor o Caracter no valido por favor, ingrese una opcion correcta:\n")
        print(repr(e))

exit(0)